### PR TITLE
Register and render the counter block in PHP

### PR DESCRIPTION
**Pull request by Gentyspun**
Ref: [#145](https://github.com/amnestywebsite/humanity-theme/issues/145)

Registers the counter block in PHP to add FSE support, also changes the render to PHP

**Steps to test**:
1.  Edit a post
2. Check if counter block appears in the block inserter
3. Add the block and content
4. Save and view the front end
5. Check block functions as expected

Example: http://bigbite.im/i/kn8U03

### CODE REVIEW

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	 * @package Amnesty/Blocks
+	 *
+	 * @return void
+	 */
+	function register_stat_counter_block() {
+		register_block_type(
+			'amnesty-core/counter',
+			[
+				'editor_script' => 'amnesty-stat-counter-block-editor',
+				'render_callback' => 'render_stat_counter_block',
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php:18]:
```suggestion
				'editor_script'   => 'amnesty-stat-counter-block-editor',
				'render_callback' => 'render_stat_counter_block',
```

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+if ( ! function_exists( 'register_stat_counter_block' ) ) {
+	/**
+	 * Register Stat Counter block
+	 *
+	 * @package Amnesty/Blocks
+	 *
+	 * @return void
+	 */
+	function register_stat_counter_block() {
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php:13]:
```suggestion
	function register_stat_counter_block(): void {
```

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare( strict_types = 1 );
+
+if ( ! function_exists( 'register_stat_counter_block' ) ) {
+	/**
+	 * Register Stat Counter block
+	 *
+	 * @package Amnesty/Blocks
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php:9]:
```suggestion
	 * @package Amnesty\Blocks
```

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare( strict_types = 1 );
+
+if ( ! function_exists( 'render_stat_counter_block' )) {
+	/**
+	 * Render Stat Counter block
+	 *
+	 * @package Amnesty/Blocks
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php:9]:
```suggestion
	 * @package Amnesty\Blocks
```

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	 *
+	 * @param array $attributes Block attributes
+	 *
+	 * @return string
+	 */
+	function render_stat_counter_block( array $attributes ): string {
+		$alignment = $attributes['alignment'] ? 'align' . $attributes['alignment'] : '';
+		$duration = $attributes['duration'] ?? 2;
+		$value = $attributes['value'] ?? '';
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php:18]:
```suggestion
		$attributes = wp_parse_args(
			[
				'alignment' => '',
				'duration'  => 2,
				'value'     => 0,
			],
			$attributes,
		);

		$alignment = 'align' . $attributes['alignment'];
		$duration  = $attributes['duration'];
		$value     = $attributes['value'];
```

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+		$wrapper_attributes = get_block_wrapper_attributes(
+			[
+				'class' => $alignment
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php:22]:
```suggestion
				'class' => $alignment,
```

**Review by jaymcp — CHANGES_REQUESTED** [review]
I'm not sure the deprecation is functioning correctly:

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php (hunk 1 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php:18]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php (hunk 3 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php:9]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php (hunk 5 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php:18]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php (hunk 6 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/stat-counter/render.php:22]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php (hunk 2 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/stat-counter/register.php]: resolved

**Pull request merged**